### PR TITLE
feat: add enabled option to Virtualizer

### PR DESCRIPTION
**Pull request by sxriff**
[Related proposal](https://github.com/TanStack/virtual/discussions/722)

### CODE REVIEW

**Comment by KevinVandy** [comment]
I'm in favor of an enable/disable option. It would really help my MRT libraries.

File: packages/virtual-core/src/index.ts [hunk 1]
```diff
@@ -328,6 +329,7 @@ export class Virtualizer<
   options!: Required<VirtualizerOptions<TScrollElement, TItemElement>>
   scrollElement: TScrollElement | null = null
   targetWindow: (Window & typeof globalThis) | null = null
+  private enabled: boolean = true
```
Comment (piecyk) [inline packages/virtual-core/src/index.ts:332]: I think we don't need an instance variable, overall we should relay on scrollElement if it's null then everything should be reseted.

enabled={false} should be same as getScrollElement: () => null

**Comment by piecyk** [comment]
@sxriff after reading the PR refactored it bit to handle reseting scroll offset, measurementsCache etc it's bit tricky when we also want to handle the initialRect, initialOffset etc, checkout the #741

**Comment by piecyk** [comment]
Thanks again for the PR @sxriff! We ended up merging another one #741 that covered this.

Closing this now. Appreciate your efforts and look forward to more contributions!

File: packages/virtual-core/src/index.ts (hunk 1 above)
Comment (sxriff) [reply packages/virtual-core/src/index.ts:332]: Yes, I guess that's a correct flow

**Comment by sxriff** [comment]
> Thanks again for the PR @sxriff! We ended up merging another one #741 that covered this.
> 
> Closing this now. Appreciate your efforts and look forward to more contributions!

@piecyk Thank you for reviewing and adding the feature, looking forward to using it in my project!

**Pull request closed without merging**